src/RemediationRoleSaver.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *      Matej Tyc <matyc@redhat.com>
+ */
+
+#include "RemediationRoleSaver.h"
+
+void RemediationSaverBase::selectFilenameAndSaveRole()
+{
+    const QString filename = QFileDialog::getSaveFileName(mParentWindow,
+        mSaveMessage.toUtf8(),
+        QString("%1.%2").arg(guessFilenameStem()).arg(mFiletypeExtension),
+        mFiletypeTemplate.arg(mFiletypeExtension), 0
+#ifndef SCAP_WORKBENCH_USE_NATIVE_FILE_DIALOGS
+        , QFileDialog::DontUseNativeDialog
+#endif
+    );
+
+    if (filename.isEmpty())
+        return;
+
+    const int result = saveToFile(filename);
+    if (result == 0)
+    {
+        // TODO: if OK
+    }
+    else
+    {
+        // TODO: if not OK
+    }
+}
+
+int RemediationSaverBase::saveToFile(const QString& filename)
+{
+    QFile outputFile(filename);
+    outputFile.open(QIODevice::WriteOnly);
+    struct xccdf_session* session = mScanningSession->getXCCDFSession();
+    struct xccdf_policy* policy = xccdf_session_get_xccdf_policy(session);
+    int result = xccdf_policy_generate_fix(policy, NULL, mFixTemplate.toUtf8(), outputFile.handle());
```

Comment:
since we are writing C++ we should handle C errors where we use C and only there IMO. Let's check the error here and raise an exception. Leaking error codes outside of methods makes it a pain to use the API.